### DISCUSSION TITLE
Import from Magenta cloud

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#96] [Q&A] [answered]
Hallo,

Bei mir klappt der Import der .kurviger Dateien nicht aus der App heraus wenn ich die Import Funktion nutze. Wenn ich aber in den Speicherplatz auf dem Handy gehe und dort die Cruiser App auswähle, klappt der Import direkt. Ist das ein Bug oder mache ich was falsch. Ich exportiere die Route ganz normal als .kurviger Datei von der Internetseite.

Hello,

For me, the import of .kurviger files does not work from the app when I use the import function. But if I go into the storage space on the phone and select the Cruiser app there, the import works directly. Is this a bug or am I doing something wrong. I export the route normally as a .curv file from the website.

_Originally posted by @ToToGSARider in https://github.com/devemux86/cruiser/discussions/33#discussioncomment-5409980_

**Comment by devemux86** [comment]
After discussion with @ToToGSARider we have this case with Magenta cloud:
- Routes can be imported when we click on the route files and select Cruiser to open them

- Routes can be imported with the app's import screen if they are on the normal file system

- Routes can not be imported with the app's import screen if they are on the Magenta cloud

@ToToGSARider thanks for the report, I will check what happens!
For reference the same behavior happens with the K1 + K2 apps.

---

Does anyone else use such cloud services (e.g. Google Drive, Dropbox, ...) and have seen anything strange on the import?

**Reply by DolomitesTom** [reply]
I have all my routes locally on the device, but also with Microsoft Onedrive it is working when I select a route from the cloud drive. It downloads it automatically and asks me with which app I want to open it. Works with Cruiser, K1 or K2.

**Reply by devemux86** [reply]
@DolomitesTom 
When you use "Routing | Import" menu to open the route from the cloud location, does it work?

**Reply by DolomitesTom** [reply]
I normally don't do so, but it also works. In the file browser I see my onedrive path, can select the right folder and open a kurviger file directly. Also nice, didn't know that this is working 😀

**Comment by devemux86** [comment]
I tested **Google Drive** (Android 13) and both processes seem to work properly:
- Click on the route file in the cloud storage and select Cruiser to open it
- Use "Routing | Import" menu and select the route in the cloud location

**Comment by ToToGSARider** [comment]
Ich habe es gerade mit der DropBox getestet. Da funktioniert es auch. Es liegt scheinbar an der Magenta Cloud. Ich weiss aber nicht was an der Cloud anders ist als an den anderen?

I just tested it with DropBox. It works there, too. It seems to be due to the Magenta Cloud. I do not know what is different about the cloud than the others?

**Reply by devemux86** [reply]
Thanks for the test, I will see if I can find an alternative way to avoid the cloud provider's special file handling.

---

Danke für den Test, ich werde sehen, ob ich einen alternativen Weg finden kann, um die spezielle Dateibehandlung des Cloud-Anbieters zu umgehen.

**Reply by ToToGSARider** [reply]
Okay, sag Bescheid wie ich dich gegebenenfalls unterstützen kann.

Okay, let me know how I can support you if necessary.

**Reply by ToToGSARider** [reply]
Hallo Emux, ich habe da was entdeckt. Wenn ich in der Magenta Cloud direkt auf die Datei drücke kommt der Fehler. Wenn ich hingegen hinten auf die Pfeile drücke, fragt er mich welches Programm. Wenn ich dann den Cruiser auswähle scheint es zu gehen. Es dauert aber einen Moment bis der Abfragedialog sich öffnet. Habe gleich wieder einen Kundentermin, würde aber später dann wieder ein Video senden.

Hello Emux, I have discovered something. If I press directly on the file in the Magenta Cloud, the error occurs. If I press on the arrows in the back, it asks me which program. If I then select the Cruiser, it seems to work. But it takes a moment until the query dialog opens. I have another customer appointment, but I would send a video later.

**Reply by ToToGSARider** [reply]
Es klappt aber nur bedingt bzw. scheinbar nicht immer. Gibt es eine Möglichkeit Log Dateien oder Debug zu machen. 

But it works only conditionally or apparently not always. Is there a possibility to make log files or debug.

**Reply by devemux86** [reply]
Thanks @ToToGSARider!

The problem begins with Android now working with streams rather than regular filenames.
So it is difficult to check the format of files. Also, some cloud providers do their own things.

I'm working on a new import process that will hopefully be much more stable in such cases.

---

Das Problem beginnt damit, dass Android jetzt mit Streams und nicht mehr mit normalen Dateinamen arbeitet.
Daher ist es schwierig, das Format von Dateien zu überprüfen. Außerdem machen einige Cloud-Anbieter ihre eigenen Sachen.

Ich arbeite an einem neuen Importprozess, der in solchen Fällen hoffentlich viel stabiler sein wird.

**Comment by devemux86** [comment]
In the next Cruiser update, we will explicitly choose the route format (gpx, itn, kurviger, mpjs) during import,
instead of the app trying to guess it from the restricted Android file system.

It should be a more stable process, hopefully improving such cases.

---

Im nächsten Cruiser-Update werden wir das Routenformat (gpx, itn, kurviger, mpjs) beim Import explizit auswählen,
anstatt dass die App versucht, es aus dem eingeschränkten Android-Dateisystem zu erraten.

Es sollte ein stabilerer Prozess sein, der solche Fälle hoffentlich verbessert.

**Reply by 0709wiwiwi** [reply]
:-((   Only one such a case  was reported ?
I NEVER noticed false imports by the comfortable Cruiser auto-selection.  
Whether it was a .kurviger or .gpx file it was always stable and correct. 
No .itn import was tested as I don't need that either. 
An extra selection to do for something that did not fail ?

**Reply by devemux86** [reply]
> I NEVER noticed false imports by the comfortable Cruiser auto-selection.

> An extra selection to do for something that did not fail ?

Until now we tried to guess the imported file format (gpx, kurviger, itn)
and then start the appropriate route file parser based on that "guess".

This was never safe, as newer Android does not like files (works with streams).
Therefore, it is safer for users to explicitly state what format they import / open.

A similar process already exists in (advanced) open .map files and .zip themes,
where the format (map or zip) is predefined. _But routes can have many formats._

**Comment by ToToGSARider** [comment]
Alles klar, ich werde dann direkt berichten wenn das Update draußen ist und ich getestet habe.

All right, I will report directly when the update is out and I have tested.

**Comment by DolomitesTom** [comment]
I guess this is an error: the Routing menu has now only Import and delete submenu in 3.0.14

**Reply by devemux86** [reply]
This is because you don't have any calculated / imported route?

The new menu are dynamic, showing only the actions that work.
Without route, there is no need to show buttons that don't work?

(same as the Map menu, when no offline maps are loaded)

**Reply by DolomitesTom** [reply]
OK thanks, understand.

**Comment by devemux86** [comment]
The new import process is available in [Cruiser 3.0.15](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

@ToToGSARider 

Can you please test this with your cloud provider?
Können Sie das bitte mit Ihrem Cloud-Anbieter testen?

**Reply by ToToGSARider** [reply]
Ja gerne, ich habe nur die Installation nicht hinbekommen. Ich habe auf den Link gedrückt und die ZIP entpackt auf dem Handy. Wo liegt die APK zur Installation?

Yes with pleasure, I just did not get the installation. I clicked on the link and unzipped the ZIP on the phone. Where is the APK to install?

**Reply by devemux86** [reply]
Android Cruiser is available on [Google Play](https://play.google.com/store/apps/details?id=gr.talent.cruiser) (it might take a while until the update appear).
[Releases](https://github.com/devemux86/cruiser/releases) page contains desktop Cruiser.

---

Android Cruiser ist auf [Google Play](https://play.google.com/store/apps/details?id=gr.talent.cruiser) verfügbar (es kann eine Weile dauern, bis das Update erscheint).
Die Seite [Releases](https://github.com/devemux86/cruiser/releases) enthält Desktop Cruiser.

**Reply by ToToGSARider** [reply]
Achso und ich habe die APK gesucht in den Dateien. Ich melde mich sobald ich getestet habe.

Oh and I have searched the APK in the files. I will report as soon as I have tested.

**Reply by 0709wiwiwi** [reply]
@ToToGSARider 
In the meantime you could just try and test that fantastic nice desktop version ?

**Reply by ToToGSARider** [reply]
Hallo zusammen, sorry, habe bis eben gearbeitet. Habe es gerade getestet und was soll ich sagen: Ich liebe es :-) funktioniert :-) 

Kommando zurück - Jetzt läuft es wieder nicht? 

Hello everyone, sorry, worked until just now. Just tested it and what can I say: I love it :-) works :-) 

Command back - Now it does not run again?

**Reply by ToToGSARider** [reply]
Wenn ich eine Datei die ich schon einmal importiert habe noch einmal importieren möchte, sagt er ungültige Datei? 

If I want to import a file I have already imported once again, it says invalid file?

**Reply by ToToGSARider** [reply]
@devemux86 ich mache eine Videoaufnahme und sende sie Dir

@devemux86 I make a video recording and send it to you

**Reply by ToToGSARider** [reply]
@devemux86 wo kann man denn den Importverlauf säubern? Scheinbar verursacht dies die Fehlermeldung wie du im Video sehen kannst.

@devemux86 where can you clean the import history? Apparently this causes the error message as you can see in the video.

**Reply by devemux86** [reply]
@ToToGSARider 

In the previous Cruiser version it did not work at all, and in the new Cruiser version it works only the first time?
Can you reproduce it continuously?

In der vorherigen Cruiser-Version hat es überhaupt nicht funktioniert, und in der neuen Cruiser-Version funktioniert es nur beim ersten Mal?
Können Sie es kontinuierlich reproduzieren?

**Reply by devemux86** [reply]
@ToToGSARider 

If you run the same test with files in normal storage (not in the cloud), does it work?

Wenn Sie denselben Test mit Dateien im normalen Speicher (nicht in der Cloud) durchführen, funktioniert er dann?

**Reply by 0709wiwiwi** [reply]
By Google Drive.
[Kurviger](https://youtube.com/shorts/Kz9u2JrVpyo)
[Gpx](https://youtube.com/shorts/sGztWIu6lko)

By import menu > G-Drive source:
[Kurviger](https://youtube.com/shorts/JRyHYgxZqHc)
[Gpx](https://youtube.com/shorts/gGguFvsV2P0)

**Reply by ToToGSARider** [reply]
@devemux86 Ja, wenn die gleiche Datei, die jetzt als ungültig gemeldet wird, im lokalen Speicher liegt, funktioniert es.  Das ist schon sehr seltsam.

@devemux86 Yes, if the same file that is now reported as invalid is in local storage, it works.  That is very strange.

**Reply by ToToGSARider** [reply]
@devemux86 I have found the bug. When you select a route in the Magenta Cloud, it is loaded offline into the cloud. If I now delete the file from the offline storage again, the import works again. Maybe something has to be taken into account? Well, otherwise I just start from the cloud.

@devemux86 Ich habe den Fehler gefunden. Wenn man eine Route in der Magenta Cloud auswählt, wird diese offline in die Cloud geladen. Wenn ich nun die Datei aus dem Offline-Speicher wieder lösche, funktioniert der Import wieder. Vielleicht muss da etwas beachtet werden? Naja, ansonsten starte ich einfach aus der Cloud.

**Comment by ToToGSARider** [comment]
@devemux86  Wenn ich die Route aus der Magenta Cloud heraus starte, läuft alles wie gewohnt.

@devemux86 When I start the route from Magenta Cloud, everything runs as usual.

**Reply by ToToGSARider** [reply]
@devemux86 ich habe den Bug gefunden. Wenn man bri der Magenta Cloud eine Toute wählt wird sie Offline in die Cloud geladen. Wenn ich die Datei jetzt aus dem Offline Speicher wieder lösche, funktioniert der Import wieder. Vielleicht muss da was berücksichtigt werden? Naja, ansonsten starte ich eben aus der Cloud.

@devemux86 I have found the bug. When I select a tote in the Magenta Cloud, it is loaded offline into the cloud. If I now delete the file from the offline storage again, the import works again. Maybe something has to be taken into account? Well, otherwise I just start from the cloud.
Greetings Toto

**Reply by devemux86** [reply]
@ToToGSARider 

There are 2 ways to import a route:
- Use the menu inside Cruiser - this works only the 1st time?
- Click a file with a file manager - this works always?

---


Es gibt 2 Möglichkeiten, eine Route zu importieren:
- Verwenden Sie das Menü in Cruiser - das funktioniert nur beim 1. Mal?
- Klicken Sie auf eine Datei mit einem Dateimanager - dies funktioniert immer?

**Reply by ToToGSARider** [reply]
@devemux86 Hoppla, habe erst jetzt erkannt, dass du mir eine Frage gestellt hattest.
Ja, beim ersten Import aus der Cloud klappt es ohne Probleme und dann nicht mehr. Aus dem Dateimanager klappt es jedes mal ohne Probleme. So nutze ich es ja seit 2020 eigentlich nur, da ich ja die gleichen Probleme in der K1 und K2 App auch hatte.

@devemux86 Oops, only now realized that you had asked me a question.
Yes, the first time I import from the cloud it works without problems and then it doesn't. From the file manager it works every time without problems. So I use it since 2020 actually only, because I had the same problems in the K1 and K2 app.

**Reply by devemux86** [reply]
Since this seems to work every time with other cloud providers (Google Drive, Dropbox, etc.),
it seems that something is going on with the Magenta Cloud?

I will see if I can find anything else to improve.

---

Da dies bei anderen Cloud-Anbietern (Google Drive, Dropbox usw.) jedes Mal zu funktionieren scheint,
mit der Magenta Cloud scheint was los zu sein?

Ich werde sehen, ob ich noch etwas anderes zur Verbesserung finden kann.

**Comment by devemux86** [comment]
@ToToGSARider 

Using the import menu in Cruiser, do not click the file to open,
can you **long press** the file and use the **Select** button of file browser?

Wenn Sie das Importmenü in Cruiser verwenden, klicken Sie nicht auf die Datei, um sie zu öffnen,
können Sie die Datei **lange drücken** und die **Auswahltaste** des Dateibrowsers verwenden?

<img src="https://user-images.githubusercontent.com/3484020/228231624-c684c71a-bee4-411c-90c6-ee3b6a711546.jpg" width="33%"/>

**Reply by ToToGSARider** [reply]
@devemux86 beim ersten mal und dann klappt auch der Import. Beim zweiten mal kommt wieder die Meldung ungültige Datei. Alles gut, Haken dran. Ich starte die Datei aus der Magenta Cloud heraus. 

@devemux86 the first time and then the import also works. The second time the message invalid file comes again. All good, hook on it. I start the file from the Magenta Cloud.

**Reply by 0709wiwiwi** [reply]
Can it be that the downloaded file is than named "xxxxx.kurviger (1)"  or xxxxx.gpx (1)

**Reply by devemux86** [reply]
> Can it be that the downloaded file is than named "xxxxx.kurviger (1)" or xxxxx.gpx (1)

The advantage of the new import (via the Cruiser menu) is that we no longer care about file extensions.
We explicitly choose the file format in the import process and the files can have any file name. :slightly_smiling_face: 

Der Vorteil des neuen Imports (über das Cruiser-Menü) ist, dass wir uns nicht mehr um Dateierweiterungen kümmern müssen.
Wir wählen das Dateiformat beim Importieren explizit aus und die Dateien können einen beliebigen Dateinamen haben. :slightly_smiling_face:

**Reply by 0709wiwiwi** [reply]
> > Can it be that the downloaded file is than named "xxxxx.kurviger (1)" or xxxxx.gpx (1)

> The advantage of the new import (via the Cruiser menu) is that we no longer care about file extensions. We explicitly choose the file format in the import process and the files can have any file name. slightly_smiling_face

_I still prefer the automatic comfortable import of previous version without the unnecessary extra selection action needed. This worked comfortably fast and fine.
It is more important that correct extensions are enforced in the produced file exports.  Especially for exchanges with external apps. Cruiser may avoid fake extensions this way, no other app imitates this.  If you start sharing file with wrong extensions it won't work in other apps.  Response: Was this produced by Cruiser ? This must be another  junk app too._

**Comment by devemux86** [comment]
The import process can be optimized much more, I have some ideas for the near future.

Der Importprozess kann noch viel mehr optimiert werden, ich habe einige Ideen für die nahe Zukunft.

**Comment by devemux86** [comment]
> ich habe den Bug gefunden. Wenn man bri der Magenta Cloud eine Toute wählt wird sie Offline in die Cloud geladen. Wenn ich die Datei jetzt aus dem Offline Speicher wieder lösche, funktioniert der Import wieder.

The import route has been rewritten in [Cruiser 3.0.16](https://play.google.com/store/apps/details?id=gr.talent.cruiser). @ToToGSARider please try again with your cloud.

I don't quite understand what you mean by deleting the file. Is the file available in the phone memory?
Or is it only available in the cloud as a link? Then it can't work.

---

Die Importroute wurde in [Cruiser 3.0.16](https://play.google.com/store/apps/details?id=gr.talent.cruiser) neu geschrieben. @ToToGSARider bitte versuchen Sie es erneut mit Ihrer Cloud.

Ich verstehe nicht ganz, was Sie mit dem Löschen der Datei meinen. Ist die Datei im Telefonspeicher verfügbar?
Oder ist sie in der Cloud nur als Link vorhanden? Dann kann es nicht funktionieren.

**Reply by ToToGSARider** [reply, marked as answer]
Es funktioniert immer noch nicht. Es ist nicht schlimm, es gibt wichtigeres. Ich denke, dass die Magenta Cloud das Problem ist. Die Route wird beim auswählen in einen Offline Speicher in der Cloud gelegt und ist dann scheinbar nicht mehr lesbar für die App. Alles gut, ich starte wenn einfach aus der Cloud heraus die Routen. In der Cloud kann ich ja auswählen mit welcher App ich die Route fahren möchte. Trotzdem Danke für den Ehrgeiz.  

It still does not work. It's not bad, there are more important things. I think that the Magenta Cloud is the problem. The route is put into an offline storage in the cloud when selected and then is apparently not readable by the app. All good, I start if simply from the cloud the routes. In the cloud I can select yes with which app I want to drive the route. Nevertheless, thanks for the ambition.  

Translated with www.DeepL.com/Translator (free version)
